app/templates/main/history.html

```diff
@@ -86,6 +86,13 @@
                 {% endfor %}
                 </tbody>
             </table>
+            <div> 
+            {% if all_info %} 
```

Comment:
Can you fix the indentation here?

Also, why are we passing `all_info` in? Why don't we just format the string for the total hours in the view function and display it or the error message here? That way we don't have so much logic in the HTML.